### PR TITLE
[UI refactor] Remove .bind() part3 - Remove .bind for SpinButton in AxisConfigDialog, MultiICEPlot and Fairness

### PR DESCRIPTION
**Pull request by tongyu-microsoft**
This PR removes .bind() for SpinButton in AxisConfigDialog, MultiICEPlot and Fairness folder. Also removes .bind() for cohortList by creating a cohortNameColumn component.

## Description
The UI refactor work is to remove all .bind() functions. And this PR is part 3.
part 1 PR link: https://github.com/microsoft/responsible-ai-toolbox/pull/1523.
part 2 PR link: https://github.com/microsoft/responsible-ai-toolbox/pull/1542.
There will be following PRs to remove other .bind() functions.
The reason that we want to remove .bind() is that, a different callback is created each time the component renders. So we want to remove .bind() and set lint rule in the end to ban adding .bind() in the future.

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: libs/core-ui/src/lib/components/AxisConfigDialogSpinButton.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export interface IAxisConfigSpinButtonProps {
+  label: string;
+  min?: number;
+  max?: number;
+  styles?: IStyleFunctionOrObject<ISpinButtonStyleProps, ISpinButtonStyles>;
+  value?: string;
+  setNumericValue: (delta: number, stringVal: string) => string | void;
+}
+
+export class AxisConfigDialogSpinButton extends React.PureComponent<IAxisConfigSpinButtonProps> {
```
Comment (imatiach-msft) [inline libs/core-ui/src/lib/components/AxisConfigDialogSpinButton.tsx:23]: nice refactor into a separate class!

**Review by imatiach-msft — APPROVED** [review]

File: libs/core-ui/src/lib/components/AxisConfigDialogSpinButton.tsx (hunk 1 above)
Comment (tongyu-microsoft) [reply libs/core-ui/src/lib/components/AxisConfigDialogSpinButton.tsx:23]: Thanks!!

File: libs/fairness/src/lib/Controls/BinDialog.styles.ts [hunk 2]
```diff
@@ -95,3 +95,17 @@ export const BinDialogStyles: () => IProcessedStyleSet<IBinDialogStyles> =
       }
     });
   };
+
+export const SpinButtonStyles = {
```
Comment (RubyZ10) [inline libs/fairness/src/lib/Controls/BinDialog.styles.ts:99]: nit: Is this the same as the one added in MultiCEPlpt.styles.ts? wondering if it makes sense to take it out and put in a common place

**Review by RubyZ10 — APPROVED** [review]

**Review by RubyZ10 — APPROVED** [review]

File: libs/fairness/src/lib/Controls/BinDialog.styles.ts (hunk 2 above)
Comment (tongyu-microsoft) [reply libs/fairness/src/lib/Controls/BinDialog.styles.ts:99]: Good point! Let me create a common SpinButtonStyles so it can be shared!

**Pull request merged**